### PR TITLE
feat(plasma-typo,plasma-core): Add duplicate some mixins for interrupt cycle in dependencies

### PR DESCRIPTION
**Pull request by neretin-trike**
:baby_chick: Download canary assets:

[colors--canary.214.3420295710.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.214.3420295710.xml)  
[color_sbermarket_ios-swift--canary.214.3420295710.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.214.3420295710.swift)  
[color_sbermarket_kotlin--canary.214.3420295710.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.214.3420295710.kt)  
[color_sbermarket_react-native--canary.214.3420295710.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.214.3420295710.ts)  
[color_sberprime_ios-swift--canary.214.3420295710.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.214.3420295710.swift)  
[color_sberprime_kotlin--canary.214.3420295710.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.214.3420295710.kt)  
[color_sberprime_react-native--canary.214.3420295710.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.214.3420295710.ts)  
[PlasmaTokensColor--canary.214.3420295710.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.214.3420295710.swift)  
[typo_mage_kotlin--canary.214.3420295710.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.214.3420295710.kt)  
[typo_mage_react-native--canary.214.3420295710.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.214.3420295710.ts)  
[typo_plasma_kotlin--canary.214.3420295710.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.214.3420295710.kt)  
[typo_plasma_react-native--canary.214.3420295710.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.214.3420295710.ts)  
[typo_ruler_kotlin--canary.214.3420295710.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.214.3420295710.kt)  
[typo_ruler_react-native--canary.214.3420295710.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.214.3420295710.ts)  
[typo_sage_kotlin--canary.214.3420295710.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.214.3420295710.kt)  
[typo_sage_react-native--canary.214.3420295710.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.214.3420295710.ts)  
[typo_sbermarket_kotlin--canary.214.3420295710.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.214.3420295710.kt)  
[typo_sbermarket_react-native--canary.214.3420295710.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.214.3420295710.ts)  
[typo_soulmate_kotlin--canary.214.3420295710.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.214.3420295710.kt)  
[typo_soulmate_react-native--canary.214.3420295710.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.214.3420295710.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.111.1-canary.214.3420295710.0
  npm install @salutejs/plasma-core@1.81.1-canary.214.3420295710.0
  npm install @salutejs/plasma-icons@1.106.1-canary.214.3420295710.0
  npm install @salutejs/plasma-temple@1.109.3-canary.214.3420295710.0
  npm install @salutejs/plasma-tokens@1.28.1-canary.214.3420295710.0
  npm install @salutejs/plasma-typo@0.28.1-canary.214.3420295710.0
  npm install @salutejs/plasma-ui@1.143.3-canary.214.3420295710.0
  npm install @salutejs/plasma-web@1.140.1-canary.214.3420295710.0
  npm install @salutejs/plasma-cy-utils@0.34.1-canary.214.3420295710.0
  npm install @salutejs/plasma-sb-utils@0.79.1-canary.214.3420295710.0
  npm install @salutejs/plasma-tokens-android@2.35.1-canary.214.3420295710.0
  npm install @salutejs/plasma-tokens-ios-swift@2.35.1-canary.214.3420295710.0
  # or 
  yarn add @salutejs/plasma-b2c@1.111.1-canary.214.3420295710.0
  yarn add @salutejs/plasma-core@1.81.1-canary.214.3420295710.0
  yarn add @salutejs/plasma-icons@1.106.1-canary.214.3420295710.0
  yarn add @salutejs/plasma-temple@1.109.3-canary.214.3420295710.0
  yarn add @salutejs/plasma-tokens@1.28.1-canary.214.3420295710.0
  yarn add @salutejs/plasma-typo@0.28.1-canary.214.3420295710.0
  yarn add @salutejs/plasma-ui@1.143.3-canary.214.3420295710.0
  yarn add @salutejs/plasma-web@1.140.1-canary.214.3420295710.0
  yarn add @salutejs/plasma-cy-utils@0.34.1-canary.214.3420295710.0
  yarn add @salutejs/plasma-sb-utils@0.79.1-canary.214.3420295710.0
  yarn add @salutejs/plasma-tokens-android@2.35.1-canary.214.3420295710.0
  yarn add @salutejs/plasma-tokens-ios-swift@2.35.1-canary.214.3420295710.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-214/

File: packages/plasma-core/src/mixins/applyHyphens.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import { css, InterpolationFunction } from 'styled-components';
 
+// NOTE: Этот модуль дублируется в @salutejs/plasma-typo для прерывания циклической зависимости
```
Comment (Yeti-or) [inline packages/plasma-core/src/mixins/applyHyphens.ts:3]: а вот меня сейчас глючит или мы можем тупо этот код импортировать из
plasma-typo, и удалить его из plasma-core?

и будет:
```
plasma-core
  <= plasm-typo
  <= cy-utils
    <= plasm-typo
```

```

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-214/
  ui storybook: http://plasma.sberdevices.ru/pr-214/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-214/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-214/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-214/temple-storybook/

File: packages/plasma-core/src/mixins/applyHyphens.ts (hunk 1 above)
Comment (neretin-trike) [reply packages/plasma-core/src/mixins/applyHyphens.ts:3]: ща проверю

Comment (neretin-trike) [reply packages/plasma-core/src/mixins/applyHyphens.ts:3]: в целом вроде как можем. Но есть момент, что эти методы также используются в plasma-web/b2c. Если мы за реэкспортим внутри plasma-core эти зависимости из plasma-typo, то не должно сломаться?

Comment (Yeti-or) [reply packages/plasma-core/src/mixins/applyHyphens.ts:3]: да конечно, api core остается

Comment (neretin-trike) [reply packages/plasma-core/src/mixins/applyHyphens.ts:3]: тогда попробую 👌  Единственное, что я не буду сейчас тащить все методы и типы для типографики, а только которые нужны в данном случае

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-214/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-214/
  ui storybook: http://plasma.sberdevices.ru/pr-214/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-214/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-214/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-214/temple-storybook/

**Review by Yeti-or — APPROVED** [review]

**Pull request merged**